atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/PreCommitCondition.java

```diff
@@ -40,6 +45,19 @@ public interface PreCommitCondition {
      */
     void throwIfConditionInvalid(long timestamp);
 
+    /**
+     * Checks that the condition is valid at the given timestamp and writes, otherwise throws a
+     * {@link TransactionFailedException}. If the condition is not valid, the transaction will not be committed.
+     * This will _only_ be called in write transactions once #commit() has been called. In all other cases,
```

Comment:
nit: `{@link SnapshotTransaction#commit()}`? Or just `Transaction`. Not sure, but check.